src/testing/mock.md

```diff
@@ -1,218 +1,310 @@
-# Mock Runtime for Testing
+# Mock Runtime for Unit Testing
+*[`kitchen/module/hello-substrate`](https://github.com/substrate-developer-hub/recipes/tree/master/kitchen/modules/hello-substrate), [`kitchen/module/adding-machine`](https://github.com/substrate-developer-hub/recipes/tree/master/kitchen/modules/adding-machine)*
 
-Runtime tests allow you to verify the logic in your runtime module by mocking a Substrate runtime environment. This requires an explicit implementation of all the traits declared in the module(s) for the mock runtime. So, if our module trait looked like,
+At the bottom of the module, we can place unit tests in a separate module with a special compilation flag
 
 ```rust
-pub trait Trait: system::Trait {
-  type Reward = SomeRewardType<AssociatedType>;
+#[cfg(test)]
+mod tests {
+	...
+}
+```
+
+To use the logic from the module to be tested, it is necessary to bring `Module` and `Trait` into scope.
+
+```rust
+use crate::{Module, Trait};
+```
+
+Now, declare the mock runtime as a blanket structure
 
-  type ConstThing = Get<u32>;
+```rust
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Runtime;
+```
+
+The `derive` macro attribute provides implementations of the `Clone + PartialEq + Eq + Debug` traits for the `Runtime` struct. 
+
+The mock runtime also needs to implement the tested module's `Trait`. 
+
+```rust
+impl Trait for Runtime {
+	type Event = TestEvent;
 }
 ```
 
-Then, the implementation would require an explicit implementation of this trait in our mock runtime (similar in structure to the [substrate](https://github.com/paritytech/substrate/tree/master/node/runtime) and [polkadot](https://github.com/paritytech/polkadot/blob/master/runtime/src/lib.rs) runtime configurations). For example,
+The `TestEvent` enum is defined to emulate the module's `Event` enum. 
 
 ```rust
-pub type SpecificType = u32; // could be some other type
+mod hello_substrate {
+	pub use crate::Event;
+}
 
-parameter_types!{
-  pub const ConstThing = 255;
+impl_outer_event! {
+	pub enum TestEvent for Runtime {
+		hello_substrate<T>,
+	}
 }
 
-impl module::Trait for Runtime {
-  type Reward = SpecificType;
-  type ConstThing = ConstThing;
+impl Trait for Runtime {
+	type Event = TestEvent;
 }
 ```
```

Comment:
I've seen `type Event = ();` a lot. It might be worth moving this bit lower to your section on testing Event Emission and mention that you only need this part of the mock if you actually want to test event emission.

Reply:
added